### PR TITLE
Fix for connector attributes not getting updated for combo changes

### PR DESCRIPTION
**Pull request by imsuneth**
Fix for https://github.com/wso2/api-manager/issues/727

### CODE REVIEW

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java [hunk 1]
```diff
@@ -83,5 +87,25 @@ private void setComponentVisibility(EnableCondition enableCondition, boolean vis
...
+    private void addToDependentControls(EnableCondition enableCondition, boolean visibility) {
+    	String targetComponentName = enableCondition.getTargetComponentName();
+        if (compositeList.containsKey(targetComponentName)) {
+            Composite composite = compositeList.get(targetComponentName);
+            if (!visibility) {
+            	if (!dependentInvisibleComposites.containsKey(composite)) {
+            		dependentInvisibleComposites.put(composite, visibility);
+            	}
+            } else {
+            	if (dependentInvisibleComposites.containsKey(composite)) {
```
Comment (SanojPunchihewa) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java:100]: Do we need this `containsKey` check?

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java [hunk 2]
```diff
@@ -83,5 +87,25 @@ private void setComponentVisibility(EnableCondition enableCondition, boolean vis
...
     }
+    
+    private void addToDependentControls(EnableCondition enableCondition, boolean visibility) {
+    	String targetComponentName = enableCondition.getTargetComponentName();
+        if (compositeList.containsKey(targetComponentName)) {
+            Composite composite = compositeList.get(targetComponentName);
+            if (!visibility) {
+            	if (!dependentInvisibleComposites.containsKey(composite)) {
+            		dependentInvisibleComposites.put(composite, visibility);
+            	}
```
Comment (SanojPunchihewa) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java:98]: Can we use `putIfAbsent`?

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java (hunk 1 above)
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java]: Fixed in https://github.com/wso2/integration-studio/pull/1167/commits/aff499587d662c8ba7e54cb69cf443367a88d7e3

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java (hunk 2 above)
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java]: Yes. Fixed in https://github.com/wso2/integration-studio/pull/1167/commits/aff499587d662c8ba7e54cb69cf443367a88d7e3

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/PropertiesWidgetProvider.java [hunk 3]
```diff
@@ -1033,6 +1035,33 @@ private void updateEnableConditionValidation(String componentName, String compon
...
+	
+	private void updateModelForInvisibleDependentControls() {
+		// update model for dependent controls
+		Set<Composite> dependentInvisibleComposites = enableConditionManager
+				.getDependentInvisibleComposites();
+
+		for (Composite composite : dependentInvisibleComposites) {
+			for (Control control : composite.getChildren()) {
+				if (control instanceof Text || control instanceof Combo || control instanceof Table) {
+					if (!isConnectionWidgetProvider) {
```
[the remark this answers is not in the document]
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/PropertiesWidgetProvider.java:1047]: yes, we can eliminate it

[the remark this answers is not in the document]
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/PropertiesWidgetProvider.java]: Fixed in https://github.com/wso2/integration-studio/pull/1167/commits/ea24a68cf666a2873d68f631ea76b8c7bbda1430

**Review by SanojPunchihewa — APPROVED** [review]

**Pull request merged**